tests/suites/test_suite_x509parse.data

```diff
@@ -1650,6 +1650,14 @@ X509 CRT ASN1 (TBS, inv v3Ext, SubjectAltName repeated outside Extensions)
 depends_on:MBEDTLS_RSA_C:MBEDTLS_SHA256_C
 x509parse_crt:"3081dc3081c6a0030201028204deadbeef300d06092a864886f70d01010b0500300c310a30080600130454657374301c170c303930313031303030303030170c303931323331323335393539300c310a30080600130454657374302a300d06092A864886F70D010101050003190030160210ffffffffffffffffffffffffffffffff0202ffffa100a200a321301f301d0603551d11041630148208666f6f2e7465737482086261722e74657374301d0603551d11041630148208666f6f2e7465737482086261722e74657374300d06092a864886f70d01010b0500030200ff":"":MBEDTLS_ERR_X509_INVALID_FORMAT + MBEDTLS_ERR_ASN1_LENGTH_MISMATCH
 
+X509 CRT ASN1 (TBS, valid v3Ext in v1 CRT)
+depends_on:MBEDTLS_RSA_C:MBEDTLS_SHA256_C
+x509parse_crt:"3081b93081a3a0030201008204deadbeef300d06092a864886f70d01010b0500300c310a30080600130454657374301c170c303930313031303030303030170c303931323331323335393539300c310a30080600130454657374302a300d06092A864886F70D010101050003190030160210ffffffffffffffffffffffffffffffff0202ffffa321301f301d0603551d11041630148208666f6f2e7465737482086261722e74657374300d06092a864886f70d01010b0500030200ff":"":MBEDTLS_ERR_X509_INVALID_FORMAT + MBEDTLS_ERR_ASN1_LENGTH_MISMATCH
```

Comment:
Why is this `MBEDTLS_ERR_ASN1_LENGTH_MISMATCH`? I am having trouble seeing where the length mismatch is. Is it from an interpretation of the ASN.1 instead of the ASN.1 itself?

Reply:
@Patater When we think we should have reached the end of an ASN.1 structure but we haven't, we return `MBEDTLS_ERR_ASN1_LENGTH_MISMATCH`, which I think is the right choice of error code for this kind of situation. The case triggers an instance of that, because the v1 CRT parser doesn't look for extensions but expects to have parsed the entire TBS body, which isn't the case if anything follows (in particular a misplaced v3 extension).

Reply:
OK, thanks for explaining. This makes sense now. It's not that any of the lengths are wrong at the ASN.1 level, but at the CRT level. Thanks!